OpenRA.Game/Map/MapPreview.cs

```diff
@@ -641,14 +608,15 @@ public void DisposePackage()
 		public void Delete()
 		{
 			Invalidate();
-			(parentPackage as IReadWritePackage)?.Delete(Package.Name);
+			(parentPackage as IReadWritePackage)?.Delete(Path);
 		}
 
 		Stream IReadOnlyFileSystem.Open(string filename)
 		{
 			// Explicit package paths never refer to a map
-			if (!filename.Contains('|') && Package.Contains(filename))
-				return Package.GetStream(filename);
+			LoadPackage();
```

Comment:
This doesn't look right. The caller should call LoadPackage themselves, and dispose afterwards

Reply:
This is the same logic that was previously being done by the `Package` property, but rewritten to make the side-effect clearer.

Changing the load/dispose behaviour would be a significant change that is way out of scope for this PR!

Reply:
Ah I see